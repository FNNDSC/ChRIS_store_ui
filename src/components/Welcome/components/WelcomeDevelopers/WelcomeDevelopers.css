.welcome-developers {
  /* display */
  padding: 6em 2em;
  padding-bottom: 8em;

  /* font */
  font-size: 1.4em;
}

.welcome-developers-header {
  /* font */
  font-size: 2em;
  line-height: 1.3;
}
