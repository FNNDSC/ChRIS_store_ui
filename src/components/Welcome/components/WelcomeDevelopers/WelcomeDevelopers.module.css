.welcome-developers {
  /* display */
  display:flex;
  flex-direction:column;
  width:70%;
  padding: 6em 2em;

  /* font */
  font-size: 1.4em;
}
.welcome-developers-header{
  /* display */
  margin-bottom: 0.5em;

  /* font */
  font-size: 2em;
  line-height: 1.3;
}
.callToAction-btn{
  width: max-content;
  margin:0 auto;
  width: 208.5px;
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 768px) {
  .welcome-developers{
    width: 70%;
  }
}