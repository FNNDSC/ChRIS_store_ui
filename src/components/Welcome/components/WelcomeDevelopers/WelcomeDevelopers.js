import React from 'react';
import './WelcomeDevelopers.css';

const WelcomeDevelopers = () => (
  <div className="welcome-developers row">
    <div className="welcome-developers-header">Developers Call To Action.</div>
  </div>
);

export default WelcomeDevelopers;
