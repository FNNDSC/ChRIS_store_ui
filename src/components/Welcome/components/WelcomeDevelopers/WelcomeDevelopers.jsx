import React from "react";
import { Grid, GridItem, TextInput } from '@patternfly/react-core';

import "./WelcomeDevelopers.css";
import Button from '../../../Button';

const WelcomeDevelopers = () => (
  <div id="welcome-developers">
    <div id="welcome-developers-header">
      <span className="tag">DEVELOPERS</span><br/>
      <h1>Expand the Reach <br/> of your Image Processing Software</h1>
      <p>
        ChRIS is an democratizing the development and usage of image processing
        <br/> software with an ecosystem following common standards.
      </p>
    </div>

    {/* <article style={{ maxWidth: '1000px' }}>
      <Grid id="welcome-developers-body">        
        <GridItem lg={12} xs={12} style={{ textAlign: 'center' }}>
          <h2 style={{ fontWeight: '600' }}>
            Open Source Software
          </h2>
          <div className="text-light">
            <p>
              <strong>ChRIS is an open source platform</strong> for medical
              analytics in the cloud, democratizing the development and usage of
              image processing software within an ecosystem following
              <strong> common standards.</strong>
            </p>
          </div>
          <Card>
            <CardBody>
              <h2 style={{ margin: '0.25em 0 1.5em 0' }}><b>Create a ChRIS Developer Account</b></h2>
              <SignUp/>
            </CardBody>
          </Card>
        </GridItem>
      </Grid>
    </article> */}

    <article>
      <Grid hasGutter id="welcome-developers-signup">
        <GridItem lg={8} lgOffset={2} xs={12}>
          <div style={{ fontWeight: 'medium', margin: '1em auto' }}>
            A developer account enables you to share your analysis
            workflows as containerized software with the <b>ChRIS</b> community of
            researchers and clinicians.
          </div>
        </GridItem>

        <GridItem lg={6} lgOffset={3} xs={12}>
        <Grid hasGutter id="call-to-action">
          <GridItem lg={8}>
          <TextInput type="email" aria-label="Email Address" id="email-address" placeholder="Email Address"/>
          </GridItem>
          <GridItem lg={4}>
          <Button variant="primary" toRoute='/quickstart'>
            Sign Up
          </Button>
          </GridItem>
          </Grid>
          <p style={{ fontSize: 'small', margin: '1em auto' }}>
            By siging-up you agree to the terms and privacy policy.
          </p>
        </GridItem>
      </Grid>
    </article>
  </div>
);

export default WelcomeDevelopers;
