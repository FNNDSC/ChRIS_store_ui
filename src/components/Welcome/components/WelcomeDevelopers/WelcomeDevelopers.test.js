import React from 'react';
import { shallow } from 'enzyme';
import WelcomeDevelopers from './WelcomeDevelopers';

describe('WelcomeDevelopers', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<WelcomeDevelopers />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render row div', () => {
    expect(wrapper.find('div.row')).toHaveLength(1);
  });

  it('should render the row div with class welcome-developers', () => {
    expect(wrapper
      .find('div.row')
      .hasClass('welcome-developers'))
      .toEqual(true);
  });

  it('should render the div with class welcome-developers-header', () => {
    expect(wrapper
      .find('div.row')
      .find('div.welcome-developers-header').length)
      .toEqual(1);
  });
});
