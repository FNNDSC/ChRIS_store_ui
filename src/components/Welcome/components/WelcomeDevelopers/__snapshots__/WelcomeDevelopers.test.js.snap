// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeDevelopers should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeDevelopers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="welcome-developers-header"
>
        Developers Call To Action.
</div>,
      "className": "welcome-developers row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Developers Call To Action.",
        "className": "welcome-developers-header",
      },
      "ref": null,
      "rendered": "Developers Call To Action.",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="welcome-developers-header"
>
          Developers Call To Action.
</div>,
        "className": "welcome-developers row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Developers Call To Action.",
          "className": "welcome-developers-header",
        },
        "ref": null,
        "rendered": "Developers Call To Action.",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
